plugins_/settings/known_settings.py

```diff
@@ -34,7 +34,7 @@ def format_completion_item(value, default=False):
                                    type(value).__name__,
                                    default_str),
             # 'cast' dicts to frozen sets, because those are hashable
-            frozenset(value.items()) if isinstance(value, dict) else value)
+            frozenset(value) if isinstance(value, dict) else value)
```

Comment:
If you don't want to try to support nested key's completion, we at most should offer the keys of the first level as completion if any at all. I think it may even not make much sense at all as those structures might differ too much in structure.

We need to ensure not to offer top level settings keys in such a value scope if no value completion is offered.